src/gui/search/searchsortmodel.cpp

```diff
@@ -29,17 +29,89 @@
 #include "searchsortmodel.h"
 
 SearchSortModel::SearchSortModel(QObject *parent)
-    : QSortFilterProxyModel(parent)
+    : base(parent)
+    , m_isNameFilterEnabled(false)
+    , m_minSeeds(0)
+    , m_maxSeeds(-1)
+    , m_minLeechs(0)
+    , m_maxLeechs(-1)
+    , m_minSize(0)
+    , m_maxSize(-1)
 {
 }
 
+void SearchSortModel::enableNameFilter(bool enable)
+{
+    m_isNameFilterEnabled = enable;
+}
+
+void SearchSortModel::setNameFilter(const QString &searchTerm)
+{
+    m_searchTerm = searchTerm;
+    if (searchTerm.length() > 2 &&
+            searchTerm.at(0) == QChar('\"') && searchTerm.at(searchTerm.length() - 1) == QChar('\"')) {
+        m_searchTermWords.append(m_searchTerm.mid(1, m_searchTerm.length() - 2));
+    }
+    else {
+        m_searchTermWords = searchTerm.split(QChar(' '), QString::SkipEmptyParts);
+    }
+}
+
+void SearchSortModel::setSizeFilter(qint64 minSize, qint64 maxSize)
+{
+    m_minSize = std::max(static_cast<qint64>(0), minSize);
```

Comment:
>GCC is right

Yes. It just can't deduce template parameter (as I said above) since it has two candidates: `int` and `qint64`. But if we set template parameter explicitly it's OK - compiler instantiates function `std::max(qint64, qint64)` and pass `0` and `minSize` to it and it uses implicit cast from `int` to `qint64` in this case.
So you can leave it as is or use `std::max<qint64>` instead.